src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -1029,6 +1029,39 @@ void Test12()
             VerifyNotInScope(model, x12Ref[1]);
         }
 
+        [Fact]
+        public void PropertyNamedInComplexPattern()
```

Comment:
This change appears to be mostly addressing IDE bugs.  Think we need to add some IDE tests to cover these scenarios.